test/test_main.py

```diff
@@ -42,13 +44,25 @@ def get_test_files():
         test_files = glob.glob(os.path.join(os.path.dirname(__file__), "files/**/*.ifc"), recursive=True)
     return test_files
 
+
+
 @pytest.mark.parametrize("filename", get_test_files())
-def test_invocation(filename):
-    results = list(run(filename))
+def test_invocation(request, filename):
+
+    def step_attrs_to_file(*attrs): 
+        def get_nested_item(d, keys):
+            for key in keys:
+                d = d.get(key, {})
+            return d
+        return list({get_nested_item(item, attrs) for item in behave_results})[0]
```

Comment:
Yes, I think so too. It would maybe be better to remove the whole logic to the protocol main package and just pass the behave_results to the function to keep it clean. Something like 
```
if '--validate-dev' in sys.argv or request.config.getoption("--validate-dev"):
   protocol.enforce(behave_results)
```